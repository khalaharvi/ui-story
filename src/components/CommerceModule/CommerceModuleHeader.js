import React from 'react';
import PropTypes from 'prop-types';

import { cx } from './CommerceModule';

const CommerceModuleHeader = ({ productName, brandName }) => (
  <div className={cx('header')}>
    <div className={cx('product-name-wrapper')}>
      <h1 itemProp="name" className={cx('product-name')}>{productName}</h1>
    </div>
    <p itemProp="brand" className={cx('brand-name')}>
      {brandName}
    </p>
  </div>
);

CommerceModuleHeader.propTypes = {
  productName: PropTypes.string.isRequired,
  brandName: PropTypes.node.isRequired
};

export default CommerceModuleHeader;
