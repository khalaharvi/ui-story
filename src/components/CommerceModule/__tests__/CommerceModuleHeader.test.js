import React from 'react';
import { shallow } from 'enzyme';

import CommerceModuleHeader from '../CommerceModuleHeader';

describe('<CommerceModuleHeader />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <CommerceModuleHeader
        productName="Product Name"
        brandName="Brand Name"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
