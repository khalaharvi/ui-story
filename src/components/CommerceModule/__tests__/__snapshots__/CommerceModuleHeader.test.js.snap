// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommerceModuleHeader /> renders correctly 1`] = `
<div
  className="dh-commerce-module-header"
>
  <div
    className="dh-commerce-module-product-name-wrapper"
  >
    <h1
      className="dh-commerce-module-product-name"
      itemProp="name"
    >
      Product Name
    </h1>
  </div>
  <p
    className="dh-commerce-module-brand-name"
    itemProp="brand"
  >
    Brand Name
  </p>
</div>
`;
